example/black_scholes.cpp

```diff
@@ -36,11 +36,11 @@ promote<Price, Sigma, Tau, Rate> black_scholes_option_price(CP cp,
   using namespace std;
   auto const d1 = (log(S / K) + (r + sigma * sigma / 2) * tau) / (sigma * sqrt(tau));
   auto const d2 = (log(S / K) + (r - sigma * sigma / 2) * tau) / (sigma * sqrt(tau));
-  switch (cp) {
-    case CP::call:
-      return S * Phi(d1) - exp(-r * tau) * K * Phi(d2);
-    case CP::put:
-      return exp(-r * tau) * K * Phi(-d2) - S * Phi(-d1);
+  if (cp == CP::call) {
+    return S * Phi(d1) - exp(-r * tau) * K * Phi(d2);
+  } else {
+    assert(cp == CP::put);
```

Comment:
Wondering why this change? I find the previous more readable.

Given the impending boost 1.71 deadline and fragility of the CI tests, I say we put this PR into the next release rather than risk additional CI issues with these multiple changes. For now I simply removed the `gamma_hpp` test template which seemed to satisfy travis. I'd also like to separate testing `autodiff` on its own, vs investigating how well `autodiff` and `boost.math` are compatible with one another. Somehow these two goals got mixed together.

Btw here is a question that may best be answered by you: https://github.com/boostorg/math/pull/222#issuecomment-506953227

Reply:
I agree regarding readability (I'd prefer a `switch` statement). The reason I moved to a conditional is that with the `switch`, is because the was being printed during compilation:
**example/black_scholes.cpp:45:1: warning: control reaches end of non-void function [-Wreturn-type]**

It's up to you how this is handled; if you want to use `switch`, I'd suggest we have throw an exception if `cp` equals neither `CP::call` nor `CP::put` (on the premise that `-Wreturn-type` is an informative warning).

Reply:
Ok thanks. I forgot that `-fstrict-enums` has to be turned on for `gcc`.

Reply:
I agree, removing the gamma function tests for this release is the best thing to do given the timeline.

The PR obviously has gamma function tests (which should work, since exceptions are caught in the case of infinitely valued `autodiff_fvar` types). `multiprecision` types also are compiling/passing for me with the gamma functions.

Bessel functions are very, very slow with `multiprecision` as `autodiff_fvar`'s real type; I need to see what's going on with those.